pkg/pdfcpu/read.go

```diff
@@ -1721,7 +1721,7 @@ func buffer(c context.Context, rd io.Reader) (buf []byte, endInd int, streamInd
 		growSize = min(growSize*2, maximumBufSize)
 		line := string(buf)
 
-		endInd, streamInd, err = model.DetectKeywords(line)
+		endInd, streamInd, err = model.DetectKeywordsWithContext(c, line)
```

Comment:
Why don't we leave the original method name DetectKeywords and pass in as 2nd parm the context from the caller. does that make any sense?